### PR TITLE
Zenodo

### LINKED ISSUE #72: Apparent authorship in Zenodo records
From to the Zenodo record (https://zenodo.org/record/5749489) it appears that all repo contributors (https://github.com/inbo/checklist/graphs/contributors) are assumed to be package author. Probably this is an unwanted feature, given the (rightfully) more limited number of authors in `DESCRIPTION` and in `.zenodo.json`.

I noticed this because of a PURE email alert.

### LINKED ISSUE #74: lintr warnings when using `check_package()`
When using function `check_package()` or the CI provided by checklist, I get >50 warnings, mostly with the same message.  The different messages are in my case:
- `Passing linters as variables was deprecated in lintr version 2.0.1.9001. Use a call to the linters (see ?linters) instead.`
- `The use of linters of class 'function' was deprecated in lintr version 2.0.1.9001. Use linters classed as 'linter' (see ?Linter) instead.`

(I know I can ignore them for now, but to resolve on occasion.  ;-) )

### PR DESCRIPTION
**Pull request by ThierryO**
fixes #74 
fixes #72

### CODE REVIEW

**Review by ElsLommelen — CHANGES_REQUESTED** [review]
Some suggestions on the vignette.  I tested all up to the title 'what happens next?', but I suppose this part will only work properly after your PR has been merged.  ;-)

File: vignettes/zenodo.Rmd
Comment (ThierryO) [reply vignettes/zenodo.Rmd:70]: My problem was that I've already configured it. And I had no clue how I did it. ;-)

So selecting "GitHub" from the profile dropdown is sufficient?

Comment (ElsLommelen) [reply vignettes/zenodo.Rmd:70]: More or less, after that I had to hit a button ('connect to github' if I remember well) and give github permission to exchange data with Zenodo. In fact similar to what I had done before for DataCite, so very straightforward, once I found the dropdown.

Actually I like your description below better (Go to the dropdown box and ...) better than:

> So selecting "GitHub" from the profile dropdown is sufficient?

The reference to 'your account (top right)' clearly indicates where to search for this less visible dropdown list, certainly for someone who visits the website for the first time.

File: vignettes/zenodo.Rmd
Comment (ElsLommelen) [reply vignettes/zenodo.Rmd:63]: Yes, much better!
I don't remember every detail, I guess I had to give a user name as well in the end, but maybe I did too much and added a Globus account while it was not needed?
In any case, this version would make it much easier to know how to start this authorisation than the first version.

File: vignettes/zenodo.Rmd
Comment (ThierryO) [reply vignettes/zenodo.Rmd]: fixes in 1873d5a

File: vignettes/zenodo.Rmd
Comment (ThierryO) [reply vignettes/zenodo.Rmd]: fixed in 1873d5a

File: vignettes/zenodo.Rmd
Comment (ThierryO) [reply vignettes/zenodo.Rmd]: I'll drop this line as it identical to the second line of the once per repo section.

File: .github/workflows/check_on_different_r_os.yml
Comment (ElsLommelen) [inline .github/workflows/check_on_different_r_os.yml:28]: So in fact you are doing twice the same check?
What I always wondered: why do you run exactly the same check on different versions of the same OS?  To truly check a package (and not just if its dependencies still run on the old version of the OS), I think it would be more useful for the 'oldrel' to check the package for older versions of the dependencies, for instance the versions that were new versions at the moment of the previous R update at INBO.

**Review by ElsLommelen — COMMENTED** [review]
I added just one small question on the ubuntu tests...

I have a PR almost ready, I suppose I have to wait until this PR is merged if I want to test if the Zenodo stuff works properly?  Then I maybe better approve here after my question is solved, and add issues afterwards if I experience problems with Zenodo?

File: .github/workflows/check_on_different_r_os.yml
Comment (ThierryO) [reply .github/workflows/check_on_different_r_os.yml:28]: Those checks are still different. `devel` refers to the daily build of R (will become `R 4.x+1`). Whereas `oldrel` refer to the previous stable minor version of R (`R 4.x-1.latest`). People no longer should be using Ubuntu 18.04. Hence there is no reason to test on 18.04. In the near future we will test `devel` on Ubuntu 22.04 and `oldrel` on Ubuntu 20.04.

**Comment by ThierryO** [comment]
> I have a PR almost ready, I suppose I have to wait until this PR is merged if I want to test if the Zenodo stuff works properly? Then I maybe better approve here after my question is solved, and add issues afterwards if I experience problems with Zenodo?

I had to push the docker manually to make this PR check. I suggest that you check inbo/inbodb#35 with this version of checklist.

**Comment by ElsLommelen** [comment]
I encounter a first problem related to trying this new version: during CI I get the [error](https://github.com/inbo/inbodb/runs/4852279915?check_suite_focus=true) that `.zenodo.json` needs an update, but locally I don't get this error.  I tried installing the newest checklist version locally using `remotes::install_github("inbo/checklist@zenodo")`, but this doesn't help: after running `checklist::check_package()` I still don't get the error locally, most likely because there is no file `.zenodo.json`.  In the NEWS file I read something about a function `write_zenodo_json()`, so I tried running this function, but still no `.zenodo.json` appears...

Honestly, to me it is also unclear how to deal with an update of checklist: do we have to run `setup_package()` for each package after each update of checklist, or is is ok to just run `check_package()` with the newest version of checklist whenever we make changes?  I always do the latter, as the documentation doesn't mention anything on it, and it seems most convenient that it just keeps working unless something is mentioned on the NEWS site.

(For clarification: I had checklist set up on this package somewhere last year, probably before the zenodo functionality was added, and up to now this error never appeared during CI, as far as I know.)

**Comment by ThierryO** [comment]
Did you restart R? The .zenodo.json file should be there as it is in inbo/inbodb#35.

Running `setup_package()` is only required when you want to use the latest GHA workflows or pkgdown styling. It doesn't hurt to run `setup_package()`. It will only rerun when the git environment is clean. The diff will tell you what was changed.

I delibrately added a session info to the output of `check_package()` so you can compare the version numbers between the local and the CI checks.

**Comment by ElsLommelen** [comment]
Ok, now I get it: RStudio doesn't show `.zenodo.json` in the explorer window.  Anyway, after restarting RStudio and opening `.zenodo.json` in notepad++ once, it appeared in the Git window after running `checklist::check_package()` again.

Yes, I already noticed the session info in the output, but for some reason I still succeed in getting different results now and then, even if the versions are the same.  I already learned I have to refresh the git window because not all file types appear automatically, apparently I also have to restart R and run the check again...

Anyway, finally my test passed, thanks for your help!  Time to check the zenodo part!

**Comment by ThierryO** [comment]
files starting with a dot are considered hidden files. They are hidden by default. You can make them visible through a setting in the file explorer or your OS.

Can you approve this PR when things run smoothly?

**Comment by ElsLommelen** [comment]
By the way, in [this section](https://inbo.github.io/checklist/articles/getting_started.html#:~:text=commit%20or%20PR.-,Finalise%20online%20setup,-Create%20a%20personal) below, there is a reference to ubuntu 18.4 'oldrel'.  So make sure you adapt this everywhere.  ;-)
(I don't know how to add it as a suggestion without starting a new branch and PR, because in this PR there are not yet any changes in this file...)

**Comment by ElsLommelen** [comment]
> files starting with a dot are considered hidden files. They are hidden by default. You can make them visible through a setting in the file explorer or your OS.

This setting is set and all hidden files are shown in Windows Explorer (and also `.zenodo.json` is shown there).  Also: other files and folders starting with a dot are shown in RStudio (`.github`, `.gitignore`, `.httr-oauth`, `.Rbuildignore`, `.Rhistory`), only `.zenodo.json` is lacking here.  (If I remember well, I have the same problem with `.env`.)  So I don't know where to solve this problem, if I just know it should be there, I know some options to open it.

**Comment by ElsLommelen** [comment]
> Can you approve this PR when things run smoothly?

I will, but only after I tested the whole process up to the publication in Zenodo.  ;-)

**Review by ElsLommelen — APPROVED** [review]
Up to now it all seems fine, but I noticed it is really necessary to follow the manuals very carefully when installing the whole stuff on a package.  It is not yet published in Zenodo, but I suppose it takes some time?  The release worked fine in any case, so I'll approve, I suppose any additional problems won't be due to the checklist package.

**Comment by ThierryO** [comment]
Did you activate the github repo at zenodo?

**Comment by ElsLommelen** [comment]
Yes, I did.
But now I noticed a mistake in my profile.  And after solving this, I noticed an error in the zenodo release, which refers to a mistake in the orcid identifier of one of the authors...  So it seems your workflow works fine, while I still have to fix some minor mistakes.  ;-)

**Comment by ElsLommelen** [comment]
Oups, apparently not a mistake in the ORCID, but the `X` in one of the orcids has changed into `x>` in the `.zenodo.json`, and I suppose this is triggering the error in Zenodo.

**Comment by ThierryO** [comment]
This is a bug in write_zenodo_json() #79. I'll fix it.

**Pull request merged**
